src/pytorch_lightning/core/mixins/device_dtype_mixin.py

```diff
@@ -12,180 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Any, Optional, Union
+from lightning_lite.utilities import rank_zero_deprecation
+from lightning_lite.utilities.device_dtype_mixin import DeviceDtypeModuleMixin as NewDeviceDtypeModuleMixin
 
-import torch
-from torch.nn import Module
-from typing_extensions import Self
 
-
-class DeviceDtypeModuleMixin(Module):
-    __jit_unused_properties__ = ["device", "dtype"]
-
-    def __init__(self) -> None:
+class DeviceDtypeModuleMixin(NewDeviceDtypeModuleMixin):
```

Comment:
can we have two commits for each file here: one that moves the entire file and one that changes imports and creates the deprecated replacement class? This would preserve git history for this file

Reply:
Oh. How important is this for us? I assumed that we would squash+merge this PR into the main one. I can pay more attention next time. Any thoughts @carmocca ?

Reply:
We can do this suggestion as long as it's a true move, meaning there won't be two copies.

But for this to work well, it would be done targeting master, not a lite/main branch. Otherwise the lite/main branch needs to be perfectly and cleanly squashable, which is probably too much of an effort

Reply:
https://github.com/Lightning-AI/lightning/pull/14511